### DISCUSSION TITLE
NumField prefix

### DISCUSSION THREAD

**Discussion by w-sr** [vazco/uniforms#989] [Q&A] [answered]
I am using uniforms for dynamic questionnaire in our project.
I need to add `$` prefix to NumField so that whenever I input amount for money, field should be displayed like `$200`.
When I try to use `TextField` for this, validation is always false because this field should be `integer`.
Is there anyway for this implementation?

**Comment by radekmie** [comment, marked as answer]
Hi @w-sr. In a way, this has nothing to do with uniforms - all you have to do is to provide a component that receives a numeric `value` and calls `onChange` with a number as well. Some libraries, like Ant Design, have such a formatting functionality built-in ([docs](https://ant.design/components/input-number/#components-input-number-demo-formatter), [uniforms playground example](https://uniforms.tools/playground/#?N4IgDgTgpgzlAuIBcIDqUA2BjA9gWygEIQAacCHMGZUAQwFd4cZaA3KZAM1ozjIaYt2AEUy0AnsgCMABhlkAJgEsWAIwxQFXHnxAZaqzMngR6UMmH1YoACxwYFUCNt7mQ0WgoDyAOwySkblcyGCwbKDxaZBAfKAB3AAIAZSU8SygksIjaACYAIQglBQBzKAAKAB0fBITYxJS0jUzwyLLgKpqayCVrJAT26s6a-HEwKD6AOXo8QwgSDqGE-h8lThwIPBg-gcXOtY3aeHgnPtYeMwSAXgA-BIADABIH4DOMMwBfO_nB3bBaCDgEFO5ygV1urzMADoYBgeuUpABKb67d7IzqohbvJFVBGkEAwOxxACSfiUsQAohAKM5Ajo3LQYOIfFhfEl6Ko8EpELTgiAGUyWT4AGo8IqHDg8uCokDwFoSvk-eBad5AA)). Material-UI does not, so you need to either implement it on your own or use an existing package. One that I found should work with Material-UI is [`react-number-format`](https://github.com/s-yadav/react-number-format). Do let us know whether that worked for you!
